chain/epoch_manager/src/lib.rs

```diff
@@ -225,11 +225,7 @@ impl EpochManager {
             let stake = epoch_info.validators[*validator_id as usize].stake;
             *versions.entry(version).or_insert(0) += stake;
         }
-        let total_block_producer_stake: u128 = epoch_info
-            .block_producers_settlement
-            .iter()
-            .map(|id| epoch_info.validators[*id as usize].stake)
-            .sum();
+        let total_block_producer_stake: u128 = epoch_info.validators.iter().map(|v| v.stake).sum();
```

Comment:
What about validators who are chunk products but not block producers?
They currently don't have a way to indicate version update, which means that it will be not transition to the new version with not enough stake in that case.

If `block_producers_settlement` has dups - then first let's build a table of ids and then sum with those ids:
```
block_producers_settlement.iter().collect::<HashSet<_>>().iter().map(|id| epoch_info.validators[*id as usize].stake).sum()
```